File: EasyAuth/src/main/java/com/mn/easyauth/EasyAuthPrinter.java
package com.mn.easyauth;

import android.util.Log;

public class EasyAuthPrinter {

    void print(String tag,String message){
        switch (EasyAuthLog.getLogType()){
            case RELEASE: printRelease(tag,message);
            case DEBUG: printDebug(tag,message);
            case ALL: printAll(tag,message);
        }
    }

    private void printDebug(String tag,String message){
        Log.d(tag,message);
    }

    private void printRelease(String tag,String message){
        Log.d(tag,message);
    }

    private void printAll(String tag,String message){
        Log.d(tag,message);
    }

}

File: EasyAuth/src/main/java/com/mn/easyauth/EasyAuthPrinter.kt
package com.mn.easyauth

import android.util.Log

class EasyAuthPrinter {
    fun print(tag: String, message: String?) {
        when (EasyAuthLog.getLogType()) {
            LogType.RELEASE -> {
                printRelease(tag, message)
                printDebug(tag, message)
                printAll(tag, message)
            }

            LogType.DEBUG -> {
                printDebug(tag, message)
                printAll(tag, message)
            }

            LogType.ALL -> printAll(tag, message)
        }
    }

    private fun printDebug(tag: String, message: String?) {
        Log.d(tag, message!!)
    }

    private fun printRelease(tag: String, message: String?) {
        Log.d(tag, message!!)
    }

    private fun printAll(tag: String, message: String?) {
        Log.d(tag, message!!)
    }
}